docs/v8-migration.md

```diff
@@ -0,0 +1,471 @@
+  In Babel 7.7.0, you can opt-in this behavior by using the `useSpread` option.
+
+- Type check input options ([#12460](https://github.com/babel/babel/pull/12460))
+
+  **Migration**: The preset will also report invalid option names. Refer to the [docs](./preset-react.md#options) and ensure valid usage.
+
+![low](https://img.shields.io/badge/risk%20of%20breakage%3F-low-yellowgreen.svg)
+
+- Disallow `filter` option in automatic runtime ([#15068](https://github.com/babel/babel/pull/15068/commits/e2dd3be6e38b0254bc69a8e52c265214235829c6))
+
+  **Migration**: The `filter` option can only be used with the `classic` runtime. If you have switched to `automatic` runtime, you can safely remove this option. Otherwise please specify `runtime: "classic"`.
+
+### `@babel/preset-typescript` {#configuration-change-preset-ts}
+
+![high](https://img.shields.io/badge/risk%20of%20breakage%3F-high-red.svg)
+
+- Remove `isTSX` and `allExtensions` option ([#14955](https://github.com/babel/babel/pull/14955))
+
+  **Migration**:
+  - `isTSX: true` and `allExtensions: true`
+
+    If you are already using `@babel/preset-react`, `@babel/plugin-transform-react-jsx` or any other third-party jsx presets such as `@vue/babel-preset-jsx`, and you want to transpile `.tsx` files, you can safely remove these two options. Babel 8 will automatically handle `.tsx` files using this preset and the other JSX plugin.
+
+    ```diff title="babel.config.json"
+    {
+      "presets": [
+        ["@babel/preset-react", { "runtime": "automatic" }],
+    -   ["@babel/preset-typescript", { "allExtensions": true, "isTSX": true }]
+    +   ["@babel/preset-typescript"]
+      ]
+    }
+    ```
+
+    If you want to transpile files other than `.tsx`, such as `.vue`, use `ignoreExtensions: true`:
+
+    ```diff title="babel.config.js"
+    {
+      overrides: [{
+        include: /\.vue$/,
+        presets: [
+          ['@babel/preset-typescript', {
+    -       allExtensions: true, isTSX: true
+    +       ignoreExtensions: true
+          }]
+        ]
+      }]
+    }
+    ```
+
+    If you want to preserve the JSX format but transpile the TypeScript part, use `ignoreExtensions: true` and add [`@babel/plugin-syntax-jsx`](./plugin-syntax-jsx.md) to `plugins`.
+
+  - `isTSX: false` and `allExtensions: true`
+
+    Use `ignoreExtensions: true`, see the example above.
+
+  - `isTSX: false` and `allExtensions: false`
+
+    You can safely remove them.
+
+
+![medium](https://img.shields.io/badge/risk%20of%20breakage%3F-medium-yellow.svg)
+
+- Remove `allowDeclareFields` option ([#12461](https://github.com/babel/babel/pull/12461))
+
+  **Migration**: Remove the option from your config. You will probably be fine with the new behaviour. Previously `allowDeclareFields` enables transforming the `declare` syntax introduced in TypeScript 3.7, in Babel 8 we support the syntax without such a flag. See also the [compilation changes](#compilation-change-jsx) section.
```

Comment:
```suggestion
  **Migration**: Remove the option from your config, since it's now enabled by default.  Previously `allowDeclareFields` enables transforming the `declare` syntax introduced in TypeScript 3.7, in Babel 8 we support the syntax without such a flag. See also the [compilation changes](#compilation-change-jsx) section.
```